base/shell.jl

```diff
@@ -255,60 +255,91 @@ shell_escape_posixly(args::AbstractString...) =
     sprint(print_shell_escaped_posixly, args...)
 
 
-function print_shell_escaped_winsomely(io::IO, args::AbstractString...)
-    first = true
-    for arg in args
-        first || write(io, ' ')
-        first = false
-        # Quote any arg that contains a whitespace (' ' or '\t') or a double quote mark '"'.
-        # It's also valid to quote an arg with just a whitespace,
-        # but the following may be 'safer', and both implementations are valid anyways.
-        quotes = any(c -> c in (' ', '\t', '"'), arg) || isempty(arg)
-        quotes && write(io, '"')
-        backslashes = 0
-        for c in arg
-            if c == '\\'
-                backslashes += 1
-            else
-                # escape all backslashes and the following double quote
-                c == '"' && (backslashes = backslashes * 2 + 1)
-                for j = 1:backslashes
-                    # backslashes aren't special here
-                    write(io, '\\')
-                end
-                backslashes = 0
-                write(io, c)
+"""
+    shell_escape_wincmd(s::AbstractString)
+    shell_escape_wincmd(io, s::AbstractString)
+
+The unexported `shell_escape_wincmd` function escapes the meta
+characters `()!^<>&|` processed by the Windows `cmd.exe` shell: it
+places a `^` in front of any metacharacter that follows an even number
+of quotation marks on the command line.
+
+The percent sign (`%`) is not escaped, therefore shell variable
+references (like `%USER%`) will still be substituted by `cmd.exe`.
+
+Input strings should avoid ASCII control characters, as many of these
+cannot be escaped (e.g., NUL, CR, LF).
+
+See also: [`escape_microsoft_c_args`](@ref), [`shell_escape_posixly`](@ref)
+
+# Example
+```jldoctest
+julia> println(shell_escape_wincmd(escape_microsoft_c_args("^\\\"^\\", "^ C")))
+"^\\\\\\"^^\\\\" "^^ C"
+```
+"""
+function shell_escape_wincmd(io, s::AbstractString)
+    # https://stackoverflow.com/a/4095133/1990689
```

Comment:
```suggestion
    # https://stackoverflow.com/a/4095133/1990689
    any(c -> c in ('\r', '\n'), s) && throw(ArgumentError("Encountered unsupported character by CMD."))
```

from: https://github.com/JuliaLang/julia/blob/09c1d61ae2ff77f53c0362a2a3cc31a2d5d95e3c/base/shell.jl#L316-L321

Reply:
Done in latest commit, although using regular expressions (benchmarks asymptotically faster) and also including `\0`.